django/db/models/sql/compiler.py

```diff
@@ -467,14 +467,21 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                         )
                     nowait = self.query.select_for_update_nowait
                     skip_locked = self.query.select_for_update_skip_locked
-                    # If it's a NOWAIT/SKIP LOCKED query but the backend
+                    of = self.query.select_for_update_of
+                    # If it's a NOWAIT/SKIP LOCKED/OF query but the backend
                     # doesn't support it, raise a DatabaseError to prevent a
                     # possible deadlock.
                     if nowait and not self.connection.features.has_select_for_update_nowait:
                         raise DatabaseError('NOWAIT is not supported on this database backend.')
                     elif skip_locked and not self.connection.features.has_select_for_update_skip_locked:
                         raise DatabaseError('SKIP LOCKED is not supported on this database backend.')
-                    for_update_part = self.connection.ops.for_update_sql(nowait=nowait, skip_locked=skip_locked)
+                    elif of and not self.connection.features.has_select_for_update_of:
+                        raise DatabaseError('FOR UPDATE OF is not supported on this database backend.')
```

Comment:
Agreed but given all the other unsupported `select_for_update()` features raise `DatabaseError` as well this could be done in a subsequent PR.